server/routes/api/landing_page.py

```diff
@@ -317,28 +321,60 @@ def scale_series(numerator, denominator):
 
     The date of the two time series may not be exactly aligned. Here we use
     year alignment to match two date. If no denominator is found for a
-    numerator, then the data is removed.
+    numerator, then the data is removed. We allow the denominator to be
+    MAX_DENOMINATOR_BACK_YEAR stale.
     """
     data = {}
+
+    num_date_fmt = None
+    denom_date_fmt = None
+    for date in denominator.keys():
```

Comment:
can we do this for just one of the keys? i imagine in most cases the format will be similar across keys. this is really expensive if the format ends up being y-m-d and it's a lot of datapoints

i'll also file a bug so we always return sorted dates from mixer to avoid doing all this sorting in python